kubespawner/spawner.py

```diff
@@ -1542,6 +1542,15 @@ def is_pod_running(self, pod):
         )
         return is_running
 
+    def is_pod_creating(self, pod):
+        """
+        Check if the given pod exists and has a UID
+
+        pod must be a dictionary representing a Pod kubernetes API object.
+        """
+
+        return pod and pod.metadata and pod.metadata.uid
+
```

Comment:
The name of this function doesn't really represent what it does. It sounds like it checks if it is in the process of being created.

Isn't it so that it will get an UID directly as part of the details you receive from the response when making the k8s API request to create the pod?

Reply:
```
        for i in range(retry_times):
            try:
                yield self.asynchronize(
                    self.api.create_namespaced_pod,
                    self.namespace,
                    pod,
                )
```

This will return the pod manifest in the end I think, which can be asserted to contain metadata.uid

Reply:
I don't believe we are guaranteed to get the UID returned in that manifest.  That code is set up as async could you clarify to me how I would get the result from that?

Reply:
If we make kubespawner stop using the old coroutines + yeild, and the new `async` and `await`, `yeild` will be replaced with `await` - which help us better understand that we will actually from this section of code where `yeild` is called, await the response, and when we have it, return it.

So like this will work I think.

```python
        for i in range(retry_times):
            try:
                created_pod = yield self.asynchronize(
                    self.api.create_namespaced_pod,
                    self.namespace,
                    pod,
                )
```